File: 0792-binary-search/0792-binary-search.rs
impl Solution {
    pub fn search(nums: Vec<i32>, target: i32) -> i32 {
        let n = nums.len();
        let (mut l, mut r) = (0, n);

        while l < r {
            let m = l + (r - l) / 2;
            if nums[m] < target {
                l = m + 1;
            } else {
                r = m;
            }
        }

        if l < n && nums[l] == target { 
            l as _ 
        } else { 
            -1 
        }
    }
}

File: 0792-binary-search/0792-binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size()-1,result = -1;
        while(l<=r){
            int m = (l+r)/2;
            if(nums[m] == target){
                result = m;
                return result;
            }
            else if(nums[m]<target){
                l = m + 1;
            }
            else{
                r = m - 1;
            }
        }
        return result;
    }
};